roscala/src/main/scala/coop/rchain/rosette/Code.scala

```diff
@@ -2,7 +2,9 @@ package coop.rchain.rosette
 
 import scala.collection.mutable
 
-case class Code(litvec: Tuple, override val _slot: mutable.Seq[Ob])
+case class Code(litvec: Tuple,
+                codevec: Seq[Op],
+                override val _slot: mutable.Seq[Ob])
```

Comment:
@pyrocto  it is because in the C++ implementation we have some operations that rely on a mutation of highly recursive constructions, something like:
```scala
def f(i: Int, ob Ob): Ob = if (i > 0) f(i-1, ob.parent) else ob
```
So it sometimes hard to keep track on this kind of mutations (also because we do not use an explicit type to track the state).

Reply:
OK; Timm, please file a ticket to understand this better and revisit it.

Reply:
Done; I see it as a temporary hack to make the translation process easier but I'm not sure if it was the right decision. Hard to tell at the moment.